--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,6 +57,8 @@
   VCS_URL https://github.com/gazebosim/${GITHUB_NAME}.git
   VCS_VERSION ${LIB_VCS_VER}
   GLOBAL_HOOK
+  PATCHES poll-serialization.patch
+  CMAKE_ARGS -DGZ_TRANSPORT_ENABLE_ZENOH=OFF
 )

 find_package(ament_cmake_test REQUIRED)
@@ -78,7 +80,7 @@
   gz_utils_vendor
 )

-if(NOT ${${LIB_NAME_FULL}_FOUND})
+if(NOT ${${LIB_NAME_FULL}_FOUND} OR FORCE_BUILD_VENDOR_PKG)
   ament_environment_hooks("${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}.dsv.in")
   # Create a dummy .sh file needed for ament_package to source the .dsv file.
   # See https://github.com/ament/ament_package/issues/145
