--- a/ros_gz_bridge/src/factory.hpp
+++ b/ros_gz_bridge/src/factory.hpp
@@ -112,10 +112,11 @@
       }
     }
 
+    auto logger = ros_node->get_logger();
     auto ros_type = ros_type_name_;
     auto gz_type = gz_type_name_;
     std::function<void(std::shared_ptr<const ROS_T>, const rclcpp::MessageInfo &)> fn =
-      [self_pub_gids, gz_pub, ros_type, gz_type, ros_node](
+      [self_pub_gids, gz_pub, ros_type, gz_type, logger](
       std::shared_ptr<const ROS_T> ros_msg,
       const rclcpp::MessageInfo & msg_info) mutable
       {
@@ -126,7 +127,7 @@
             return;
           }
         }
-        ros_callback(ros_msg, gz_pub, ros_type, gz_type, ros_node);
+        ros_callback(ros_msg, gz_pub, ros_type, gz_type, logger);
       };
 
     auto options = rclcpp::SubscriptionOptions();
@@ -151,9 +152,9 @@
       return;
     }
     std::function<void(const GZ_T &)> subCb =
-      [this, pub, gz_to_ros_parameters](const GZ_T & _msg)
-      {
-        this->gz_callback(_msg, pub, gz_to_ros_parameters);
+      [pub, gz_to_ros_parameters](const GZ_T & _msg)
+      {
+        Factory<ROS_T, GZ_T>::gz_callback(_msg, pub, gz_to_ros_parameters);
       };
 
     // Ignore messages that are published from this bridge.
@@ -169,13 +170,13 @@
     gz::transport::Node::Publisher & gz_pub,
     const std::string & ros_type_name,
     const std::string & gz_type_name,
-    rclcpp::Node::SharedPtr ros_node)
+    const rclcpp::Logger & logger)
   {
     GZ_T gz_msg;
     convert_ros_to_gz(*ros_msg, gz_msg);
     gz_pub.Publish(gz_msg);
     RCLCPP_INFO_ONCE(
-      ros_node->get_logger(),
+      logger,
       "Passing message from ROS %s to Gazebo %s (showing msg only once per type)",
       ros_type_name.c_str(), gz_type_name.c_str());
   }
